subprojects/dependency-management/src/main/java/org/gradle/api/internal/attributes/DefaultAttributesSchema.java

```diff
@@ -53,6 +55,7 @@
     private final IsolatableFactory isolatableFactory;
     private final Map<ExtraAttributesEntry, Attribute<?>[]> extraAttributesCache = Maps.newHashMap();
     private final List<AttributeDescriber> consumerAttributeDescribers = Lists.newArrayList();
+    private final List<Attribute<?>> precedence = Lists.newArrayList();
```

Comment:
nit: replace RHS with `new ArrayList<>();`

I know it's consistent with the line above, but diamond syntax is better here and doesn't rely on guava.

Even guava's javadoc says so:
> Note for Java 7 and later: this method is now unnecessary and should be treated as deprecated. Instead, use the ArrayList constructor directly, taking advantage of the new "diamond" syntax .